File: frontend/event_planr/lib/domain/event/models/event.dart
import 'package:equatable/equatable.dart';
import 'package:event_planr/domain/event/models/event_category.dart';
import 'package:flutter/material.dart';

@immutable
class Event extends Equatable {
  const Event({
    required this.id,
    required this.name,
    required this.category,
    required this.venue,
    required this.fromDate,
    required this.coverImageUrl,
  });

  final String id;
  final String name;
  final EventCategory category;
  final String venue;
  final DateTime fromDate;
  final ImageProvider coverImageUrl;

  @override
  List<Object?> get props => [
        id,
        name,
        category,
        venue,
        fromDate,
        coverImageUrl,
      ];
}

File: backend/Shared/EventPlanR.Domain/Entities/Event.cs
using EventPlanr.Domain.Common;
using EventPlanr.Domain.Enums;

namespace EventPlanr.Domain.Entities;

public class Event : BaseAuditableEntity
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
    public EventCategory Category { get; set; }
    public DateTimeOffset FromDate { get; set; }
    public DateTimeOffset ToDate { get; set; }
    public string Venue { get; set; } = null!;
    public Address Address { get; set; } = null!;
    public Coordinates Coordinates { get; set; } = null!;
    public Language Language { get; set; }
    public Currency Currency { get; set; }
    public bool IsPrivate { get; set; }
    public Guid OrganizationId { get; set; }
    public Organization Organization { get; set; } = null!;
    public List<NewsPost> NewsPosts { get; set; } = new List<NewsPost>();
    public List<Ticket> Tickets { get; set; } = new List<Ticket>();
}
